net/api/test_server.cc.h

```diff
@@ -9,17 +9,216 @@
 #ifndef BRICKS_NET_API_TEST_SERVER_CC_H
 #define BRICKS_NET_API_TEST_SERVER_CC_H
 
-#include "../../dflags/dflags.h"
+#include <thread>
 
+#include "api.h"
+#include "../url/url.h"
 #include "../http/test_server.cc.h"
 
+#include "../../strings/printf.h"
+
 namespace bricks {
 namespace net {
 namespace api {
 namespace test {
 
-struct TestHTTPServer_APIImpl : bricks::net::http::test::TestHTTPServer_HTTPImpl {
-  // TODO(dkorolev): Real API implementation goes here.
+enum class Method { UNINITIALIZED, GET, POST };
+
+struct Request {
+  const url::URL& url;
+  HTTPServerConnection& connection;
+  const HTTPReceivedMessage& message;
+};
+
+struct Routes {
```

Comment:
Routes is not the best name I suppose. In NGINX they are called locations, in golang - handlers, may be there are better names.